src/experiments.rs

```diff
@@ -195,49 +195,122 @@ impl Experiment {
             return Ok(Some((false, experiment)));
         }
 
-        // First look at experiments explicitly assigned to this agent, and then
-        // distributed experiments and experiments without an assigned agent.
-        Experiment::next_inner(db, Some(assignee))
+        // Get an experiment whose requirements are met by this agent, preferring (in order of
+        // importance):
+        //    - experiments that were explicitly assigned to us.
+        //    - distributed experiments.
+        //    - experiments with a higher priority.
+        //    - older experiments.
+        Experiment::next_inner(db, Some(assignee), assignee)
             .and_then(|ex| {
                 ex.map_or_else(
-                    || Experiment::next_inner(db, Some(&Assignee::Distributed)),
+                    || Experiment::next_inner(db, Some(&Assignee::Distributed), assignee),
+                    |exp| Ok(Some(exp)),
+                )
+            })
+            .and_then(|ex| {
+                ex.map_or_else(
+                    || Experiment::next_inner(db, None, assignee),
                     |exp| Ok(Some(exp)),
                 )
             })
-            .and_then(|ex| ex.map_or_else(|| Experiment::next_inner(db, None), |exp| Ok(Some(exp))))
     }
 
+    #[allow(unreachable_code)]
```

Comment:
What is unreachable here?

Reply:
the piece of code relative to the cli query is after `unimplemented!`.

Reply:
Could you put a comment explaining that?

Reply:
sure